gobblin-compaction/src/main/java/org/apache/gobblin/compaction/mapreduce/CompactionJobConfigurator.java

```diff
@@ -90,6 +92,11 @@
   @Getter
   protected Collection<Path> mapReduceInputPaths = null;
   @Getter
+  protected Collection<Path> oldFiles = null;
```

Comment:
Two things here: 
1. Can you add some comments on why these things are needed. 
2. Is that possible a dataset have tons of files ? are there concerns on using in-memory data structure here